### DISCUSSION TITLE
Aysnc columnDef meta fields?

### DISCUSSION THREAD

**Discussion by callmekatootie** [TanStack/table#5263] [Q&A]
In my ColumnDefs, I am using the meta attribute to specify that a column needs to be a select dropdown.

The options for this select need to come from an API.

Are there any examples of using the ColumnDef meta object to fetch data asynchronously, before I can use it my cell() function?

Or is there a different approach I need to take?

**Comment by KevinVandy** [comment]
You can just useState for dynamic values used by the Column Def. Just remember to update the useMemo dependency array
